http_server/request_testing/request_test.go

```diff
@@ -0,0 +1,112 @@
+package requesttesting
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"testing"
+)
+
+type handlerRecorder struct {
+	req         *http.Request
+	finishedReq chan bool
```

Comment:
channels that are used to signal termination are normally named "doneSomething", in this case doneReq.